### PR TITLE
[CI] fix dockerfile for Ubuntu 20.04 support

### PR DESCRIPTION
**Pull request by junliume**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -59,6 +59,7 @@ DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
     rocm-opencl \
     rocm-opencl-dev \
     rocblas \
+    rpm \
```
Comment (shurale-nkn) [inline Dockerfile:62]: why do we need it?

**Review by shurale-nkn — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (junliume) [reply Dockerfile:62]: @shurale-nkn  it is to get rid of a warning message at `cmake` stage, mostly cosmetic I agree

Comment (junliume) [reply Dockerfile:62]: @shurale-nkn  I am okay with removing it for now (see no harm having it as is too), but we need this PR or #1503 in `develop` branch or that branch is blocked by CI base docker update :)
We need either of these two as quickly as possible

File: Dockerfile [hunk 2]
```diff
@@ -126,10 +128,9 @@ ARG MIOTENSILE_VER="default"
...
-RUN  wget -O ck.tar.gz https://www.github.com/rocmsoftwareplatform/composable_kernel/archive/${CK_COMMIT}.tar.gz && \
-    tar zxvf ck.tar.gz &&\
-    cd composable_kernel-${CK_COMMIT} && \
-    mkdir build && cd build && \
+RUN rm -rf /tmp/ck* && mkdir /tmp/ck && wget -O ck.tar.gz https://www.github.com/rocmsoftwareplatform/composable_kernel/archive/${CK_COMMIT}.tar.gz -O /tmp/ck.tar.gz && \
+    tar zxvf /tmp/ck.tar.gz -C /tmp/ && mkdir /tmp/composable_kernel-${CK_COMMIT}/build &&\
+    cd /tmp/composable_kernel-${CK_COMMIT}/build && \
     CXX=/opt/rocm/bin/hipcc cmake -DCMAKE_CXX_COMPILER_LAUNCHER="${COMPILER_LAUNCHER}" -DCMAKE_PREFIX_PATH=/opt/rocm -D CMAKE_CXX_FLAGS=" --offload-arch=gfx900 --offload-arch=gfx906 --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx1030 -O3 " .. && \
-    make -j $(nproc) install 
+    make -j $(nproc) install && rm -rf /tmp/*
```
Comment (junliume) [inline Dockerfile:135]: This is to save docker image size (saved ~5 GB)

File: Dockerfile [hunk 3]
```diff
@@ -13,25 +27,11 @@ echo $ROCM_APT_VER &&\
...
-apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 9386B48A1A693C5C && \
-wget -q -O - https://repo.radeon.com/rocm/rocm.gpg.key | apt-key add - && \
-wget --no-check-certificate -qO - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | apt-key add - && \
-sh -c "echo deb https://apt.kitware.com/ubuntu/ bionic main | tee -a /etc/apt/sources.list" && \
-apt-get update && \
 DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
     build-essential \
-    cmake-data=3.15.1-0kitware1 \
-    cmake=3.15.1-0kitware1 \
+    cmake \
```
Comment (junliume) [inline Dockerfile:34]: cmake version requirement is met by Ubuntu 20.04

File: Dockerfile [hunk 4]
```diff
@@ -5,6 +5,20 @@ ARG USE_MLIR="OFF"
 # Support multiarch
 RUN dpkg --add-architecture i386
 
+# Install preliminary dependencies
```
Comment (junliume) [inline Dockerfile:8]: Moving it here to get rid of some warning messages in cmake stage

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -247,6 +247,11 @@ if(MIOPEN_TEST_GFX900 OR MIOPEN_TEST_GFX906 OR MIOPEN_TEST_GFX908)
     list(APPEND SKIP_TESTS test_bn_3d_spatial_test)
 endif()
 
+#TODO WORKAROUND_ISSUE_1743: issue with GemmFwdRest
```
Comment (junliume) [inline test/CMakeLists.txt:250]: ref to https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1745
should remove as soon as the CI docker is updated again

File: src/comgr.cpp [hunk 6]
```diff
@@ -1251,7 +1251,7 @@ class HiprtcProgram
                 return {error_handling ? "warning: HIPRTC error log empty" : ""};
             std::vector<char> buffer(n);
             HIPRTC_CALL_INFO_THROW(hiprtcGetProgramLog(prog.get(), buffer.data()), n);
-            assert(buffer.back() == 0);
+            assert(buffer.back() == 0 || buffer.back() == '\n' || buffer.back() == '\0');
```
Comment (junliume) [inline src/comgr.cpp:1254]: ref: https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1648#issuecomment-1242357421

**Review by shurale-nkn — APPROVED** [review]
LGTM

**Pull request merged**